modules/context/context.go

```diff
@@ -154,15 +157,52 @@ func Contexter() macaron.Handler {
 			csrf:    x,
 			Flash:   f,
 			Session: sess,
+			Link:    setting.AppSubURL + strings.TrimSuffix(c.Req.URL.Path, "/"),
 			Repo: &Repository{
 				PullRequest: &PullRequest{},
 			},
 			Org: &Organization{},
 		}
-		// Compute current URL for real-time change language.
-		ctx.Data["Link"] = setting.AppSubURL + strings.TrimSuffix(ctx.Req.URL.Path, "/")
-
+		c.Data["Link"] = ctx.Link
 		ctx.Data["PageStartTime"] = time.Now()
+		// Quick responses appropriate go-get meta with status 200
+		// regardless of if user have access to the repository,
+		// or the repository does not exist at all.
+		// This is particular a workaround for "go get" command which does not respect
+		// .netrc file.
+		if ctx.Query("go-get") == "1" {
+			ownerName := c.Params(":username")
+			repoName := c.Params(":reponame")
+			branchName := "master"
+
+			owner, err := models.GetUserByName(ownerName)
+			if err != nil {
```

Comment:
Seems to be fetching the default branch-name, so `if err { branchName = "master" }` ?